include/seqan3/std/span

```diff
@@ -348,6 +358,16 @@ public:
     constexpr index_type size_bytes() const noexcept { return size_ * sizeof(element_type); }
     constexpr bool empty()            const noexcept { return size_ == 0; }
 
+    constexpr reference front() const noexcept
+    {
+        return this->operator[](0);
```

Comment:
wouldn't this be more intuitive?

```suggestion
        return this->operator[0];
```

Reply:
but this does not work. Only this would work `(*this)[0]` AFAIK but I think it is not more readable.

Reply:
Ah I thought that you tried this.

We have different styles right now:

https://github.com/seqan/seqan3/blob/a7278bbb55800ae9216d27c2212984aa92b57022/include/seqan3/core/char_operations/predicate_detail.hpp#L49-L55

https://github.com/seqan/seqan3/blob/a7278bbb55800ae9216d27c2212984aa92b57022/include/seqan3/range/views/take.hpp#L273-L280

(I only found these occurrences)

Far more common in our code base is:

https://github.com/seqan/seqan3/blob/a7278bbb55800ae9216d27c2212984aa92b57022/include/seqan3/io/alignment_file/sam_tag_dictionary.hpp#L350-L360

https://github.com/seqan/seqan3/blob/a7278bbb55800ae9216d27c2212984aa92b57022/include/seqan3/range/container/bitcompressed_vector.hpp#L425-L432

https://github.com/seqan/seqan3/blob/a7278bbb55800ae9216d27c2212984aa92b57022/include/seqan3/range/container/concatenated_sequences.hpp#L526-L532

https://github.com/seqan/seqan3/blob/a7278bbb55800ae9216d27c2212984aa92b57022/include/seqan3/range/container/dynamic_bitset.hpp#L974-L980

https://github.com/seqan/seqan3/blob/a7278bbb55800ae9216d27c2212984aa92b57022/include/seqan3/range/container/small_vector.hpp#L369-L376

https://github.com/seqan/seqan3/blob/a7278bbb55800ae9216d27c2212984aa92b57022/include/seqan3/range/decorator/gap_decorator.hpp#L368-L373

Reply:
So what should I do now? I think calling out the operator helps to understand what is meant.  But I don't really care. Also keep in mind that this is just inside the std module which is kind of external and not necessarily applying to all our rules.

Reply:
Let's keep your way, I just wanted to say that we use the other style more often and that it is not less readable.